src/main/java/org/elasticsearch/index/query/MatchAllQueryBuilder.java

```diff
@@ -100,15 +114,42 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                 }
             }
         }
+    }
 
-        if (boost == 1.0f) {
-            return Queries.newMatchAllQuery();
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        this.boost = in.readFloat();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeFloat(this.boost);
+    }
+    
+    @Override
+    public int hashCode() {
+        int hash = super.hashCode();
+        hash = maybeHashcode(hash, boost);
```

Comment:
This is copy from Lees PR, happy to simplify this in any way. Isabel mentioned utility functions in some google lib we are already using for this, will ask her about it.